bears/general/InvalidLinkBear.py

```diff
@@ -80,14 +82,22 @@ def find_links_in_file(file, network_timeout, link_ignore_regex,
                                         # Unbalanced parenthesis
             (?<!\.)(?<!,)               # Exclude trailing `.` or `,` from URL
             """, re.VERBOSE)
+        context = {}
         for line_number, line in enumerate(file):
             match = regex.search(line)
             if match:
                 link = match.group()
+                if not context.get(link):
+                    context[link] = enum(xml_namespace=False)
+                xmlns_match = xmlns_regex.search(line)
+                if xmlns_match and link in xmlns_match.groups():
+                    context[link].xml_namespace = True
                 if not (link_ignore_regex.search(link) or
                         fnmatch(link, link_ignore_list)):
                     if link.startswith(('hg+', 'bzr+', 'git+', 'svn+')):
-                        link = InvalidLinkBear.parse_pip_vcs_url(link)
+                        parsed_link = InvalidLinkBear.parse_pip_vcs_url(link)
```

Comment:
just a note : this can become another context , e.g. "pip_vcs", in the future (not this PR).